### PR TITLE
Save db before query

### PR DESCRIPTION
**Pull request by cage2**
Hi @bastibe! How you doing?

There is not so much in this patch but I found annoying the searching the annotation's database would ignore the annotated buffers that have annotations not yet saved to disk. So I ensured all the annotations in buffer are dumped to db before performing the query. I am quite sure this was my intention when i wrote the code, some time ago, but I just wrote it wrong. :)

Bye!
C.

### CODE REVIEW

File: README.org [hunk 1]
```diff
@@ -90,42 +76,24 @@ Important note: if ~/home/user/foo.notes~ exists, *will be overwritten*.
...
-    ~annotate-endline-annotate-whole-line~ is not nil (default is non
-    nil) the whole line is annotated (or the next if the line is
-    empty).
+If point is the newline character and the customizable variable ~annotate-endline-annotate-whole-line~ is not nil (default is non nil) the whole line is annotated (or the next if the line is empty).
 
-    If the line contains a single annotation that cover all the line
-    annotating the newline Will ask to edit the annotation. If
-    ~annotate-endline-annotate-whole-line~ is nil annotating a newline
-    will signal an error.
+If the line contains a single annotation that cover all the line, annotating the newline will ask to edit the annotation. If ~annotate-endline-annotate-whole-line~ is nil annotating a newline will signal an error.
```
Comment (cage2) [inline README.org:92]: this is the only non cosmetic fix to readme

Comment (cage2) [reply README.org:92]: and now that i reread the changes i think it could be considered cosmetic as well :)

File: README.org [hunk 2]
```diff
@@ -319,18 +243,16 @@ buffer) beyond the current cursor position.
...
 #+BEGIN_SRC text
  [file-mask] [(and | or) [not] regex-note [(and | or) [not] regexp-note ...]]
 #+END_SRC
 
-where
+where:
 
   - file-mask :: is a regular expression that should match the path of file
-    the annotation refers to;
+the annotation refers to;
```
Comment (bastibe) [inline README.org]: This change seems to break the markup, at least in Github's little preview.

**Comment by bastibe** [comment]
Hi @cage2,

I'm doing well, thanks for asking! Finally settling into a groove in our new home, life's been treating us well. How about you?

The pull request looks good! As always!

File: README.org (hunk 2 above)
Comment (cage2) [reply README.org]: Thanks @bastibe !

**Comment by cage2** [comment]
> Hi @cage2,

Hi!

> I'm doing well, thanks for asking! Finally settling into a groove in our new home, life's been treating us well.

I am happy to know that you are fine in your new home, changing home is a big source of stress (at least for me always has been :-)).

>  How about you?

Not too bad, but the summer has started and the temperature keep raising!  😓😓😓

Bye!
C.

**Pull request merged**